compiler/noirc_evaluator/src/ssa/opt/flatten_cfg.rs

```diff
@@ -307,8 +313,23 @@ impl<'f> Context<'f> {
         })
     }
 
+    /// Use the provided map to say if the instruction is a call to a no_predicates function
+    fn is_no_predicate(
+        &self,
+        no_predicates: &HashMap<FunctionId, bool>,
+        instruction: &InstructionId,
+    ) -> bool {
+        let mut result = false;
+        if let Instruction::Call { func, .. } = self.inserter.function.dfg[*instruction] {
+            if let Value::Function(fid) = self.inserter.function.dfg[func] {
+                result = *no_predicates.get(&fid).unwrap_or(&false);
+            }
+        }
+        result
+    }
+
     // Inline all instructions from the given block into the entry block, and track slice capacities
-    fn inline_block(&mut self, block: BasicBlockId) {
+    fn inline_block(&mut self, block: BasicBlockId, no_predicates: &HashMap<FunctionId, bool>) {
```

Comment:
Nit: I think it would be a bit cleaner to have this map be part of the `Context`, but this is non-blocking as we don't have to thread the map through many functions.

Reply:
That's what I did initially but they were borrow issues so passing the map is what seems the simplest to me.